### DISCUSSION TITLE
Intercepted routes performance issue/loading.js not working

### DISCUSSION THREAD

**Discussion by nickythorne** [vercel/next.js#66838] [App Router]
I have an intercepted parallel route, which uses a RSC page. In production and development, event with prefetching enabled the RSC network call takes 2-3seconds to complete before the route loads. 

After adding a `loading.tsx` at the same level as the parallel route, it seems like this does not help.

My structure is as follows:

```
root/
    @modal/
        (.)child/
            [child-id]        
                page.tsx <- RSC
                loading.tsx
    layout.tsx <- renders @modal slot
    page.tsx
    child/
        [child-id]
            page.tsx <- page being intercepted    
```

I have a link which navigates to the intercepted route:

```
<Link href={`./child/123?${searchParams ? searchParams.toString() : ''}`} scroll={false}>
    <Button type="secondary" size="sm">
        <Pencil2Icon />
        Review
    </Button>
</Link>
```

But in develoment and production builds, when clicking this link there is a RSC call for the route that takes 2-3seconds before the modal appears. I would expect the loading.tsx file to show instantly given the page is suspending, but this is not the case.

**Comment by tfmcmahon** [comment]
I'm having a similar issue with a similar setup. Do loading components work with intercepted routes?
